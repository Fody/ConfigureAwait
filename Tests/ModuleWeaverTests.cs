using System.Reflection;
using Fody;

public partial class ModuleWeaverTests
{
    static TestResult testResult;

    static ModuleWeaverTests()
    {
        Assembly.Load("xunit.assert");
        var weaver = new ModuleWeaver();
        testResult = weaver.ExecuteTestRun("AssemblyToProcess.dll");
        VerifierSettings.UniqueForRuntime();
        VerifierSettings.UniqueForAssemblyConfiguration();
    }
}