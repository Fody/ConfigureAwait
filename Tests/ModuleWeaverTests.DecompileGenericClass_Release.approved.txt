.class private auto ansi sealed beforefieldinit AssemblyToProcess.GenericClass`1<TItem>
extends [System.Runtime]System.Object
{
.class auto ansi sealed nested private beforefieldinit '<Method>d__0'<TItem>
extends [System.Runtime]System.ValueType
implements [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine
{
.field public int32 '<>1__state'
.field public valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder '<>t__builder'
.field public class [System.Runtime]System.Threading.Tasks.Task`1<!TItem> itemTask
.field private valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!TItem> '<>u__1'
.method private hidebysig newslot virtual final
instance void  MoveNext() cil managed
{
.override [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine::MoveNext
.maxstack  3
.locals init (int32 V_0,
valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!TItem> V_1,
valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1<!TItem> V_2,
class [System.Runtime]System.Exception V_3)
IL_0000:  ldarg.0
IL_0001:  ldfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>1__state'
IL_0006:  stloc.0
.try
{
IL_0007:  ldloc.0
IL_0008:  brfalse.s  IL_0048
IL_000a:  ldarg.0
IL_000b:  ldfld      class [System.Runtime]System.Threading.Tasks.Task`1<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::itemTask
IL_0010:  ldc.i4.0
IL_0011:  callvirt   instance valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1<!0> class [netstandard]System.Threading.Tasks.Task`1<!TItem>::ConfigureAwait(bool)
IL_0016:  stloc.2
IL_0017:  ldloca.s   V_2
IL_0019:  call       instance valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!0> valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1<!TItem>::GetAwaiter()
IL_001e:  stloc.1
IL_001f:  ldloca.s   V_1
IL_0021:  call       instance bool valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!TItem>::get_IsCompleted()
IL_0026:  brtrue.s   IL_0064
IL_0028:  ldarg.0
IL_0029:  ldc.i4.0
IL_002a:  dup
IL_002b:  stloc.0
IL_002c:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>1__state'
IL_0031:  ldarg.0
IL_0032:  ldloc.1
IL_0033:  stfld      valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>u__1'
IL_0038:  ldarg.0
IL_0039:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>t__builder'
IL_003e:  ldloca.s   V_1
IL_0040:  ldarg.0
IL_0041:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::AwaitUnsafeOnCompleted<valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!TItem>,valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>>(!!0&,
!!1&)
IL_0046:  leave.s    IL_0098
IL_0048:  ldarg.0
IL_0049:  ldfld      valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>u__1'
IL_004e:  stloc.1
IL_004f:  ldarg.0
IL_0050:  ldflda     valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>u__1'
IL_0055:  initobj    valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!TItem>
IL_005b:  ldarg.0
IL_005c:  ldc.i4.m1
IL_005d:  dup
IL_005e:  stloc.0
IL_005f:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>1__state'
IL_0064:  ldloca.s   V_1
IL_0066:  call       instance !0 valuetype [netstandard]System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1/ConfiguredTaskAwaiter<!TItem>::GetResult()
IL_006b:  pop
IL_006c:  leave.s    IL_0085
}  // end .try
catch [System.Runtime]System.Exception
{
IL_006e:  stloc.3
IL_006f:  ldarg.0
IL_0070:  ldc.i4.s   -2
IL_0072:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>1__state'
IL_0077:  ldarg.0
IL_0078:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>t__builder'
IL_007d:  ldloc.3
IL_007e:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::SetException(class [System.Runtime]System.Exception)
IL_0083:  leave.s    IL_0098
}  // end handler
IL_0085:  ldarg.0
IL_0086:  ldc.i4.s   -2
IL_0088:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>1__state'
IL_008d:  ldarg.0
IL_008e:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>t__builder'
IL_0093:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::SetResult()
IL_0098:  ret
}
.method private hidebysig newslot virtual final
instance void  SetStateMachine(class [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine stateMachine) cil managed
{
.override [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine::SetStateMachine
.maxstack  8
IL_0000:  ldarg.0
IL_0001:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>t__builder'
IL_0006:  ldarg.1
IL_0007:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::SetStateMachine(class [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine)
IL_000c:  ret
}
}
.class auto ansi sealed nested private beforefieldinit '<Method_WithValueTask>d__1'<TItem>
extends [System.Runtime]System.ValueType
implements [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine
{
.field public int32 '<>1__state'
.field public valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder '<>t__builder'
.field public class [System.Runtime]System.Threading.Tasks.Task`1<!TItem> itemTask
.field private valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!TItem> '<>u__1'
.method private hidebysig newslot virtual final
instance void  MoveNext() cil managed
{
.override [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine::MoveNext
.maxstack  3
.locals init (int32 V_0,
valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!TItem> V_1,
valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<!TItem> V_2,
valuetype [System.Threading.Tasks.Extensions]System.Threading.Tasks.ValueTask`1<!TItem> V_3,
class [System.Runtime]System.Exception V_4)
IL_0000:  ldarg.0
IL_0001:  ldfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>1__state'
IL_0006:  stloc.0
.try
{
IL_0007:  ldloc.0
IL_0008:  brfalse.s  IL_0050
IL_000a:  ldarg.0
IL_000b:  ldfld      class [System.Runtime]System.Threading.Tasks.Task`1<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::itemTask
IL_0010:  call       instance void valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<!TItem>::.ctor(valuetype [System.Private.CoreLib]System.Threading.Tasks.ValueTask`1<!0>)
IL_0015:  stloc.3
IL_0016:  ldloca.s   V_3
IL_0018:  ldc.i4.0
IL_0019:  callvirt   instance valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<!0> valuetype [System.Private.CoreLib]System.Threading.Tasks.ValueTask`1<!TItem>::ConfigureAwait(bool)
IL_001e:  stloc.2
IL_001f:  ldloca.s   V_2
IL_0021:  call       instance valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!0> valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1<!TItem>::GetAwaiter()
IL_0026:  stloc.1
IL_0027:  ldloca.s   V_1
IL_0029:  call       instance bool valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!TItem>::get_IsCompleted()
IL_002e:  brtrue.s   IL_006c
IL_0030:  ldarg.0
IL_0031:  ldc.i4.0
IL_0032:  dup
IL_0033:  stloc.0
IL_0034:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>1__state'
IL_0039:  ldarg.0
IL_003a:  ldloc.1
IL_003b:  stfld      valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>u__1'
IL_0040:  ldarg.0
IL_0041:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>t__builder'
IL_0046:  ldloca.s   V_1
IL_0048:  ldarg.0
IL_0049:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::AwaitUnsafeOnCompleted<valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!TItem>,valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>>(!!0&,
!!1&)
IL_004e:  leave.s    IL_00a2
IL_0050:  ldarg.0
IL_0051:  ldfld      valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>u__1'
IL_0056:  stloc.1
IL_0057:  ldarg.0
IL_0058:  ldflda     valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>u__1'
IL_005d:  initobj    valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!TItem>
IL_0063:  ldarg.0
IL_0064:  ldc.i4.m1
IL_0065:  dup
IL_0066:  stloc.0
IL_0067:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>1__state'
IL_006c:  ldloca.s   V_1
IL_006e:  call       instance !0 valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1/ConfiguredValueTaskAwaiter<!TItem>::GetResult()
IL_0073:  pop
IL_0074:  leave.s    IL_008f
}  // end .try
catch [System.Runtime]System.Exception
{
IL_0076:  stloc.s    V_4
IL_0078:  ldarg.0
IL_0079:  ldc.i4.s   -2
IL_007b:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>1__state'
IL_0080:  ldarg.0
IL_0081:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>t__builder'
IL_0086:  ldloc.s    V_4
IL_0088:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::SetException(class [System.Runtime]System.Exception)
IL_008d:  leave.s    IL_00a2
}  // end handler
IL_008f:  ldarg.0
IL_0090:  ldc.i4.s   -2
IL_0092:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>1__state'
IL_0097:  ldarg.0
IL_0098:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>t__builder'
IL_009d:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::SetResult()
IL_00a2:  ret
}
.method private hidebysig newslot virtual final
instance void  SetStateMachine(class [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine stateMachine) cil managed
{
.override [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine::SetStateMachine
.maxstack  8
IL_0000:  ldarg.0
IL_0001:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>t__builder'
IL_0006:  ldarg.1
IL_0007:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::SetStateMachine(class [System.Threading.Tasks]System.Runtime.CompilerServices.IAsyncStateMachine)
IL_000c:  ret
}
}
.method public hidebysig instance class [System.Runtime]System.Threading.Tasks.Task
Method(class [System.Runtime]System.Threading.Tasks.Task`1<!TItem> itemTask) cil managed
{
63 65 73 73 2E 47 65 6E 65 72 69 63 43 6C 61 73   // cess.GenericClas
73 60 31 2B 3C 4D 65 74 68 6F 64 3E 64 5F 5F 30   // s`1+<Method>d__0
00 00 )
.maxstack  2
.locals init (valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem> V_0,
valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder V_1)
IL_0000:  ldloca.s   V_0
IL_0002:  ldarg.1
IL_0003:  stfld      class [System.Runtime]System.Threading.Tasks.Task`1<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::itemTask
IL_0008:  ldloca.s   V_0
IL_000a:  call       valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::Create()
IL_000f:  stfld      valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>t__builder'
IL_0014:  ldloca.s   V_0
IL_0016:  ldc.i4.m1
IL_0017:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>1__state'
IL_001c:  ldloc.0
IL_001d:  ldfld      valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>t__builder'
IL_0022:  stloc.1
IL_0023:  ldloca.s   V_1
IL_0025:  ldloca.s   V_0
IL_0027:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::Start<valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>>(!!0&)
IL_002c:  ldloca.s   V_0
IL_002e:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method>d__0'<!TItem>::'<>t__builder'
IL_0033:  call       instance class [System.Runtime]System.Threading.Tasks.Task [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::get_Task()
IL_0038:  ret
}
.method public hidebysig instance class [System.Runtime]System.Threading.Tasks.Task
Method_WithValueTask(class [System.Runtime]System.Threading.Tasks.Task`1<!TItem> itemTask) cil managed
{
63 65 73 73 2E 47 65 6E 65 72 69 63 43 6C 61 73   // cess.GenericClas
73 60 31 2B 3C 4D 65 74 68 6F 64 5F 57 69 74 68   // s`1+<Method_With
56 61 6C 75 65 54 61 73 6B 3E 64 5F 5F 31 00 00 ) // ValueTask>d__1..
.maxstack  2
.locals init (valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem> V_0,
valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder V_1)
IL_0000:  ldloca.s   V_0
IL_0002:  ldarg.1
IL_0003:  stfld      class [System.Runtime]System.Threading.Tasks.Task`1<!0> valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::itemTask
IL_0008:  ldloca.s   V_0
IL_000a:  call       valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::Create()
IL_000f:  stfld      valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>t__builder'
IL_0014:  ldloca.s   V_0
IL_0016:  ldc.i4.m1
IL_0017:  stfld      int32 valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>1__state'
IL_001c:  ldloc.0
IL_001d:  ldfld      valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>t__builder'
IL_0022:  stloc.1
IL_0023:  ldloca.s   V_1
IL_0025:  ldloca.s   V_0
IL_0027:  call       instance void [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::Start<valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>>(!!0&)
IL_002c:  ldloca.s   V_0
IL_002e:  ldflda     valuetype [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder valuetype AssemblyToProcess.GenericClass`1/'<Method_WithValueTask>d__1'<!TItem>::'<>t__builder'
IL_0033:  call       instance class [System.Runtime]System.Threading.Tasks.Task [System.Threading.Tasks]System.Runtime.CompilerServices.AsyncTaskMethodBuilder::get_Task()
IL_0038:  ret
}
.method public hidebysig specialname rtspecialname
instance void  .ctor() cil managed
{
.maxstack  8
IL_0000:  ldarg.0
IL_0001:  call       instance void [System.Runtime]System.Object::.ctor()
IL_0006:  ret
}
}
